web/extensions/core/maskeditor.js

```diff
@@ -0,0 +1,519 @@
+		button.style.left = left;
+		return button;
+	}
+
+	createRightButton(name,right,callback) {
+		var button = this.createButton(name,callback);
+		button.style.right = right;
+		return button;
+	}
+
+	createLeftSlider(self, name,left,callback) {
+		const divElement = document.createElement('div');
+		divElement.style.position = "absolute";
+		divElement.style.top = "5px";
+		divElement.style.left = left;
+		divElement.style.backgroundColor = 'Black';
+		divElement.style.paddingLeft = "10px";
+		divElement.style.paddingRight = "10px";
+		divElement.style.borderRadius = "10px";
+
+		self.brush_slider_input = document.createElement('input');
+		self.brush_slider_input.setAttribute('type', 'range');
+		self.brush_slider_input.setAttribute('min', '1');
+		self.brush_slider_input.setAttribute('max', '100');
+		self.brush_slider_input.setAttribute('value', '10');
+
+		const labelElement = document.createElement("label");
+		labelElement.innerHTML = `<font color='white' size='5px'>${name}</font>`;
+
+		divElement.appendChild(labelElement);
+		divElement.appendChild(self.brush_slider_input);
+
+		self.brush_slider_input.addEventListener("change", callback);
+
+		return divElement;
+	}
+
+	setlayout(imgCanvas, maskCanvas) {
+		const self = this;
+
+		// If it is specified as relative, using it only as a hidden placeholder for padding is recommended
+		// to prevent anomalies where it exceeds a certain size and goes outside of the window.
+		var placeholder = document.createElement("div");
+		placeholder.style.position = "relative";
+		placeholder.style.height = "50px";
+
+		var bottom_panel = document.createElement("div");
+		bottom_panel.style.position = "absolute";
+		bottom_panel.style.bottom = "0px";
+		bottom_panel.style.left = "0";
+		bottom_panel.style.right = "0";
+		bottom_panel.style.height = "50px";
+
+		var clearButton = this.createLeftButton("Clear", "280px",
+			() => {
+				self.maskCtx.clearRect(0, 0, self.maskCanvas.width, self.maskCanvas.height);
+				self.backupCtx.clearRect(0, 0, self.backupCanvas.width, self.backupCanvas.height);
+			});
+
+		var saveButton = this.createRightButton("Save", "110px", () => {
+			document.removeEventListener("mouseup", MaskEditorDialog.handleMouseUp);
+			document.removeEventListener("keyup", MaskEditorDialog.handleKeyUp);
+				self.save();
+			});
+
+		var cancelButton = this.createRightButton("Cancel", "20px", () => {
+			document.removeEventListener("mouseup", MaskEditorDialog.handleMouseUp);
+			document.removeEventListener("keyup", MaskEditorDialog.handleKeyUp);
+			self.close();
+		});
+
+		var brush_size_slider = this.createLeftSlider(self, "Thickness", "20px", (event) => {
+			self.brush_size = event.target.value;
+		});
+
+		this.element.appendChild(imgCanvas);
+		this.element.appendChild(maskCanvas);
+		this.element.appendChild(placeholder); // must below z-index than bottom_panel to avoid covering button
+		this.element.appendChild(bottom_panel);
```

Comment:
```suggestion
	        var brush = document.createElement("div");
        brush.id = "brush";
        brush.style.backgroundColor = "transparent";
        brush.style.border = "1px dashed black";       
        brush.style.borderRadius = "50%";
        brush.style.MozBorderRadius = "50%";
        brush.style.WebkitBorderRadius = "50%";
        brush.style.position = "absolute";
        brush.style.zIndex = 100;
        brush.style.pointerEvents = "none";
        this.brush = brush;

		this.element.appendChild(imgCanvas);
		this.element.appendChild(maskCanvas);
		this.element.appendChild(placeholder); // must below z-index than bottom_panel to avoid covering button
		this.element.appendChild(bottom_panel);
        document.body.appendChild(brush);
```